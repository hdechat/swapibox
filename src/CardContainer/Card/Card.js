import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({ card }) => {
  return (
    <article>
     <h2>{card.name}</h2>
     <h5>Species: {card.species}</h5>
     <h5>Homeworld: {card.homeworld}</h5>
     <p>pop: {card.population}</p>

    </article>
  )
}

Card.propTypes = {
  card: PropTypes.object
}

export default Card;