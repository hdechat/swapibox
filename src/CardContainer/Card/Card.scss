article {
  border: solid;
  height: 260px;
  margin: 0.5rem;
  width: 155px;
}