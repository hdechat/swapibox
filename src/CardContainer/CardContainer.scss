section {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: auto;
  width: 75%;
  height: 300px;
  overflow-y: scroll;
}