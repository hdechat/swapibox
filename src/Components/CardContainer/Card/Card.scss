article {
  border: solid;
  margin: 0.5rem;
}

h6 {
  font-size: 16px;
  margin: 1rem .5rem;
}

img {
  margin: .1rem;
  width: 15%;
}

img.active {
  background: red;
  border-radius: 50%;
}

img.not-active {
  border-radius: 50%;
}

