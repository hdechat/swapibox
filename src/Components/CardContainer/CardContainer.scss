section {
  border: solid;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 1rem auto;
  width: 75%;
  height: 500px;
  overflow-y: scroll;
}

.card-container {
  border: none;
  border-style: inset;
}

h1 {
  font-size: 50px;
}

.no-cards {
  background: black;
  color: #feda4a;
  display: block;
  font-size: 400%;
  height: 75px;
  letter-spacing: 0.1em;
  margin-top: 10rem;
}

.cards-to-display {
  display: none;
}

