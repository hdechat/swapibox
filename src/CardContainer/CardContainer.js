import React from 'react';
import PropTypes from 'prop-types';
import Card from './Card/Card.js';
import './CardContainer.css';

const CardContainer = ({ cardList, clickedCard }) => {
  const cards = cardList.map((card, index) => {
    return(
      <Card key={index} card={card} clickedCard={clickedCard}/>
    )
  });

  return(
    <section>
      {cards}
    </section>
  )
}

CardContainer.propTypes = {
  cardList: PropTypes.array
}

export default CardContainer;