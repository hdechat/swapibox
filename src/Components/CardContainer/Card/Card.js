import React from 'react'
import PropTypes from 'prop-types'
import './Card.css'
import favoriteButton from '../../../assets/logo.png'
// import favorited from '../../../assets/logo-red.png'

export default function Card ({ card, clickedCard }) {
  const data = Object.entries(card).map(([key, value], index) => {
    if (key !== 'favorited') {
      if(index === Object.entries(card).length - 1) {
        return (
          <p key={value + index}>{key}: {value}</p>
        ) 
      } else {
        return (
          <div className="card-item">
            <h6 className="card-item-key" key={value + index}>{key}:</h6>
            <h6 className="card-item-value">{value}</h6>
          </div>
        )
    }
  }
})

  return (
    <article>
      <img
        onClick={() => clickedCard(!card.favorited, card)}
        className={card.favorited ? 'active' : ''}
        src={favoriteButton}
        alt="Star Wars Rebel Alliance Logo"
      />
      {data}
    </article>
  )
}

Card.propTypes = {
  card: PropTypes.object.isRequired,
  clickedCard: PropTypes.func.isRequired
}
