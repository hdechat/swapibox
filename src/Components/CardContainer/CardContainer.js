import React from 'react'
import PropTypes from 'prop-types'
import Card from './Card/Card.js'
import './CardContainer.css'

const CardContainer = ({ cardList, clickedCard }) => {
  const cards = cardList.map((card, index) => {
    return (
      <Card key={index} card={card} clickedCard={clickedCard}/>
    )
  })

  return (
    <section>
      <h1 className={cardList.length ? 'cards-to-display' : 'no-cards'}>
      There Are No Cards To Display</h1>
      {cards}
    </section>
  )
}

CardContainer.propTypes = {
  cardList: PropTypes.array.isRequired
}

export default CardContainer
