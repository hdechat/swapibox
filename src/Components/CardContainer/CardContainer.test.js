import React from 'react';
import reactDom from 'react-dom';
import { shallow, mount } from 'enzyme';
import Card from './Card/Card.js'
import CardContainer from './CardContainer.js'

describe('CardContainer', () => {
  let cardContainer;
  let mockClickedCard = jest.fn()
  let mockArray = [
    {name: 'Name', planet: 'Earth', favorited: false},
    {name: 'Name2', planet: 'Mars', favorited: false}
  ];

  beforeEach(() => {
    cardContainer = shallow(<CardContainer clickedCard={mockClickedCard} cardList={mockArray}/>)
  });

  it('matches snapshot', () => {
    expect(cardContainer).toMatchSnapshot();
  });

  it('renders each card in props array', () => {
    const expectation = mockArray.length
    expect(cardContainer.find(Card).length).toEqual(expectation);
  });
});

  //**Functional, need to learn testing for this
  // it('renders alternate text only if there are no cards to display', () => {
  //   mockArray = []
  //   expect(cardContainer.find('h1').length).toEqual(1)

  //   mockArray = [
  //   {name: 'Name', planet: 'Earth', favorited: false},
  //   {name: 'Name2', planet: 'Mars', favorited: false}
  //   ]
  //   expect(cardContainer.find('h1').length).toEqual(0)    
  // });