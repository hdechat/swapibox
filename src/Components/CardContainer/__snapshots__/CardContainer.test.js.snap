// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cardList={
        Array [
            Object {
              "favorited": false,
              "name": "Name",
              "planet": "Earth",
            },
            Object {
              "favorited": false,
              "name": "Name2",
              "planet": "Mars",
            },
          ]
    }
    clickedCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="cards-to-display"
>
          There Are No Cards To Display
</h1>,
        Array [
          <Card
            card={
                        Object {
                                    "favorited": false,
                                    "name": "Name",
                                    "planet": "Earth",
                                  }
            }
            clickedCard={[Function]}
/>,
          <Card
            card={
                        Object {
                                    "favorited": false,
                                    "name": "Name2",
                                    "planet": "Mars",
                                  }
            }
            clickedCard={[Function]}
/>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "There Are No Cards To Display",
          "className": "cards-to-display",
        },
        "ref": null,
        "rendered": "There Are No Cards To Display",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "favorited": false,
            "name": "Name",
            "planet": "Earth",
          },
          "clickedCard": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "favorited": false,
            "name": "Name2",
            "planet": "Mars",
          },
          "clickedCard": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="cards-to-display"
>
            There Are No Cards To Display
</h1>,
          Array [
            <Card
              card={
                            Object {
                                          "favorited": false,
                                          "name": "Name",
                                          "planet": "Earth",
                                        }
              }
              clickedCard={[Function]}
/>,
            <Card
              card={
                            Object {
                                          "favorited": false,
                                          "name": "Name2",
                                          "planet": "Mars",
                                        }
              }
              clickedCard={[Function]}
/>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "There Are No Cards To Display",
            "className": "cards-to-display",
          },
          "ref": null,
          "rendered": "There Are No Cards To Display",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "favorited": false,
              "name": "Name",
              "planet": "Earth",
            },
            "clickedCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "favorited": false,
              "name": "Name2",
              "planet": "Mars",
            },
            "clickedCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
